### ISSUE TITLE
Android - HSBC UK banking app will not load unless FUTO keyboard is disabled/removed

### ISSUE THREAD

**Issue by studioeng** [futo-org/android-keyboard#773]
### What happened?

When launching the HSBC UK banking app, a 'security risk' pop up screen appears. Only option is to close the app. Disabling the FUTO keyboard option and using (shudder) GBoard allows the all to work. For reference OpenBoard and SimpleBoard also result in the same dialog.

![HSBC_UK_Security_Warning](https://github.com/user-attachments/assets/a843d052-04fc-40f0-a4bc-65fba0af8639)

### Keyboard Version

0.1.24-playstore

**Comment by dean-futo** [comment]
@abb128 this may be an issue with banking apps in general, especially since some have their own keyboards they make you use for security purposes.

**Comment by abb128** [comment]
It's a known issue that some banking apps have a limited whitelist of keyboards which they allow, typically including only the most popular keyboards. There is not much we can do on our end to resolve this issue; the bank needs to update their whitelist to include FUTO Keyboard. You could request your bank to allow FUTO Keyboard if you'd like to see this resolved. In the meantime you could either access your bank through their website or switch to another keyboard when opening the bank app.

**Comment by chileverde** [comment]
In the meantime you might try the keyboard supplied with Keepass2Android Offline (info at https://philipp.crocoll.net/keepass2android/index.php). You would have to download the app to get the keyboard, but you don't have to use it as a pw manager. Once the keyboard is installed on your phone, it appears in the list of keyboards as Keepass2Android Offline and can be used with other apps. Since it has been around for a while, maybe your bank's app supports it.

**Comment by studioeng** [comment]
I had a sense that you might not have any real control over what they deem 'safe', but I thought I would put it on the radar incase it was a possible fix.

**Comment by abb128** [comment]
If you're an HSBC customer you can try submitting a complaint or feedback here: https://www.hsbc.co.uk/help/feedback-and-complaints/

**Comment by studioeng** [comment]
Ahh, brilliant. I shall do that and report back if I hear anything.

**Comment by studioeng** [comment]
Just tried again with 'Keepass2Android Offline' specifically and see the same message.

**Comment by chileverde** [comment]
Sorry to hear my idea didn't work.

Hope you can get HSBC to supply a list of allowed keyboards (maybe one doesn't spy) and/or add FUTO to the list.

**Comment by juniper-oly** [comment]
The other option is could futo switch to the default for some apps, so delegate in instances when this happens so users don't have to switch ultimately most will likely just remove the keyboard so they can use the apps they need to.

If I can use futo and it can switch out to a different keyboard for certain apps that would be great.

**Comment by dean-futo** [comment]
A user recently reported a similar issue: https://discord.com/channels/1224276522848751646/1224276522848751649/1352234743319756872

**Comment by huevos-y-bacon** [comment]
I know this is an old issue but obviously still live. Today I finally installed the app after I watched Peter's release video, but i got this when trying to open by credit card app (Marks' & Spencer Bank, UK) 
<img width="520" height="681" alt="Image" src="https://github.com/user-attachments/assets/356a4743-5715-4e7d-b327-aff02e460e98" />

**Comment by markdibley** [comment]
Still an issue. FUTO needs to approach the bank and provide the paperwork that demonstrates the required security. Until then it makes using FUTO an inconvenience.

**Comment by Ljuk67** [comment]
Same for the First Direct bank app (UK).

**Comment by abb128** [comment]
I will repeat my earlier response that the best way to get this issue resolved is if multiple customers are filing complaints to the bank about this issue. There is nothing we can do at a technical level to resolve this issue, as the banks themselves need to update their whitelist or get their vendors to do so. So if you're taking the time to post here please make sure you're also taking the time to inform your bank.

**Comment by Ljuk67** [comment]
Fair point, will do!

**Comment by kazink** [comment]
I also have this issues with First Direct bank's app (branch of HSBC). I'm in the process of filing a complaint with them, but I wanted to note something funny. If I disable the FUTO Keyboard, then open the FD app, it opens fine. I can then switch to settings, enable FUTO again, then switch to FD app and switch the keyboard to FUTO (in the help chat). The app doesn't complain 😀. I guess it will if I close the app and run it again, but it's funny how they care so much about keyboard security, that they check it only on startup.

**Comment by DrLegitamate** [comment]
I've just started experiencing this on my banking app

**Comment by oniony** [comment]
To me, at least, it's not acceptable for my bank to dictate what apps I can and cannot have on my own phone. This "security" feature was implemented unannounced and with no "accept the risk and continue" option. I can't even log on to the web as you need a code from the phone app to do this.

Ideally I would close the account and move on, but reasons. So I instead phoned them up and requested an old skool physical key and have uninstalled the app. It was either that or a second phone for bullshit apps to live on.

**Comment by kazink** [comment]
> To me, at least, it's not acceptable for my bank to dictate what apps I can and cannot have on my own phone. This "security" feature was implemented unannounced and with no "accept the risk and continue" option. I can't even log on to the web as you need a code from the phone app to do this.

Despite what the message suggests, the bank app does not require that the FUTO keyboard is uninstalled completely from the phone. It just needs to not be selected as a default keyboard when the app is started. Have the AOSP keyboard for that purpose, that I switch to in another app, and then start the bank app, then switch back to FUTO. It's inconvenient, but less inconvenient that calling them or using another phone 😀

**Comment by haritvsmurali** [comment]
you don't actually have to disable FUTO, but can just switch to Gboard using keyboard change shortcut button quickly. once banking app is closed you can switch back to FUTO again. 

This is really frustrating, as I want to Disable Gboard fully and only for this reason I've kept it enabled.. 🫠

**Comment by sarahsquirrel** [comment]
Thinking about the cybersecurity standards around a banking app: what info are they likely to look at when deciding whether to whitelist a keyboard?  

I understand that you can't change what they whitelist, but in theory you could approach HSBC and ask them what info you can deliver to support their security auditing and provide them assurances? 

Using HSBC on the web is not a viable workaround as launching the app is required for 2FA in most scenarios. Contacting the bank to provide feedback also requires logging on (i.e. launching the app)

**Comment by markdibley** [comment]
Yeah, I agree Sarah. That would be a sensible thing to do, but the impression I get from the developers is they can't really be bothered.

**Comment by kazink** [comment]
It's the banks, not (only) developers. The answer I've got from HSBC was along the lines of "we can't review every one of the thousands keyboards apps out there". So my guess is the keyboard would only be reviewed by the banks if they were one of the huge companies (Google, Apple, etc.), or perhaps had hundreds of thousands of users of the keyboard.
